src/app/clusters/door-lock-server/door-lock-server.cpp

```diff
@@ -114,7 +114,20 @@ bool DoorLockServer::SetLockState(chip::EndpointId endpointId, DlLockState newLo
     // DlLockState::kNotFullyLocked has no appropriate event to send. Also it is unclear whether
     // it should schedule auto-relocking. So skip it here. Check for supported states explicitly
     // to handle possible enum extending in future.
-    VerifyOrReturnError(DlLockState::kLocked == newLockState || DlLockState::kUnlocked == newLockState, success);
+    VerifyOrReturnError(DlLockState::kLocked == newLockState || DlLockState::kUnlocked == newLockState ||
+                            DlLockState::kUnlatched == newLockState,
+                        success);
+
+    if (DlLockState::kUnlatched == newLockState)
+    {
+        SendLockOperationEvent(endpointId, LockOperationTypeEnum::kUnlatch, opSource, OperationErrorEnum::kUnspecified, userIndex,
+                               Nullable<chip::FabricIndex>(), Nullable<chip::NodeId>(), credentials, success);
+
+        success = SetLockState(endpointId, DlLockState::kUnlocked);
```

Comment:
Wait, why?  I would expect the actual lock to do that SetLockState, when it releases the latch!